src/components/PopupWithConfirmation.js

```diff
@@ -0,0 +1,30 @@
+import Popup from "./Popup.js";
+
+export default class PopupWithFormSubmit extends Popup {
+  constructor(popupSelector) {
+    super(popupSelector);
+    this._popupForm = this._popupElement.querySelector(".modal__form");
+    this._submitButton = this._popupForm.querySelector(
+      ".modal__button_confirm"
+    );
+    this._submitButtonValue = this._submitButton.textContent;
+  }
+
+  setSubmitAction(action) {
+    this._submitAction = action;
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`